runtime/interpreter/value.go

```diff
@@ -656,8 +658,6 @@ func NewDeferredArrayValue(
 	version uint16,
 ) *ArrayValue {
 	return &ArrayValue{
-		// TODO: type
-		Type:            nil,
```

Comment:
I added this TODO so we wouldn't forget to add it here. 
Or will the type never be available here and later lazily decoded?

Reply:
Yeah, it will be added later, when the value is lazily decoded. I added it in #1035, so removed it from here.